ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java

```diff
@@ -30,491 +33,266 @@
+        addOption(LOG_TRACE, "log-trace");
+        addOption(PRINT_VERSION, "version");
+        addOption(VIEW_TYPE, "view_type");
+        addOption(RETRIES, "retries");
+        addOption(BUFFER_SIZE, "buffer_size");
+
         this.processCommandLine();
     }
 
-    private void processCommandLine() throws ParseException, BadArgumentException {
+    private String[] rootArgumentsOnly() throws BadArgumentException {
+        // Present only COMMAND, help and version arguments to parser, After the COMMAND
+        // is instantiated, the full parse will be run against the complete Options list.
+        final List<String> rootArguments = new ArrayList<String>();
+
+        for (int i = 0; i < this.args.length; i++) {
+            final String token = this.args[i];
+            // allow get COMMAND (-c) and subsequent argument
+            // or --output-format and subsequent argument
+            if (matchesOption(COMMAND, token) || matchesOption(VIEW_TYPE,  token) ||
+                    matchesOption(ACCESS_KEY, token) || matchesOption(SECRET_KEY,  token) ||
+                    matchesOption(ENDPOINT, token) || matchesOption(PROXY,  token) ||
+                    matchesOption(RETRIES, token)) {
+                rootArguments.add(token);
+                if (i < this.args.length) {
+                    rootArguments.add(this.args[++i]);
+                } else {
+                    throw new BadArgumentException("No argument supplied for option: " + token);
+                }
+            }
+            // allow get COMMAND help and subsequent argument (unless it is followed by an option (- or --)
+            if (matchesOption(COMMAND_HELP, token)) {
+                rootArguments.add(token);
+                // might or might not have arg
+                if ((i < this.args.length) && (!this.args[i + 1].startsWith("-"))) {
+                    rootArguments.add(this.args[++i]);
+                }
+            }
+            // allow version in root parse
+            if ((matchesOption(VERBOSE, token)) || (matchesOption(PRINT_HELP, token))
+                    || (matchesOption(TRACE, token)) || (matchesOption(DEBUG, token)) || (matchesOption(VERBOSE, token))
+                    || (matchesOption(LOG_TRACE, token)) || (matchesOption(LOG_DEBUG, token))
+                    || (matchesOption(LOG_VERBOSE, token))) {
+                rootArguments.add(token);
+            }
+        }
+
+        // create a String[] of only valid root parse args.
+        return rootArguments.toArray(new String[rootArguments.size()]);
+    }
+
+    /**
+     * call after adding all options to reparse
+     */
+    public void parseCommandLine() throws ParseException, BadArgumentException  {
+        parseCommandLine(false);
+    }
+
+    // parse command line. Set isRootParse true to first filter for only first parse options
+    private void parseCommandLine(final boolean isRootParse) throws ParseException, BadArgumentException {
```

Comment:
Rather than having a boolean dictate the behavior of this command, have 2 different methods, `parseRootOptions` and `parseCommandOptions`